reader/src/main/java/org/kaazing/monitoring/reader/impl/MetricsCollectorAgrona.java

```diff
@@ -41,19 +41,20 @@ public MetricsCollectorAgrona(CountersManagerEx countersManager) {
     }
 
     @Override
-    public List<Metric> getMetrics() {
-        List<Metric> metrics = new ArrayList<Metric>();
+    public List<Counter> getCounters() {
+        List<Counter> counters = new ArrayList<Counter>();
```

Comment:
We can't afford to have this method create objects because since the method is called repeatedly it will cause a lot of garbage collection and reduce performance. Here's a proposal to avoid any object creation:
- create the list of counters in the constructor and store it in a member variable
- getCounters then just returns this member variable
- You will need to change CounterImpl so it can dynamically read the latest value from valuesBuffer. That can be done by changing the constructor to the following:
```
  CounterImpl(String label, int id, CountersManagerEx countersManager)
```
 and having the getValue method do 
```
return countersManager.getLongValueForId(id);
```